packages/foreman/foreman-fapolicyd/foreman-fapolicyd.spec

```diff
@@ -28,11 +28,23 @@ cp 61-foreman-proxy.rules %{buildroot}%{_sysconfdir}/fapolicyd/rules.d/61-forema
 
 %post
 
-%{_sbindir}/fagenrules --load
+if %{_bindir}/systemctl is-active --quiet fapolicyd.service; then
+  %{_sbindir}/fagenrules --load
+  # loading fagenrules leaves fapolicyd.service in a dead state, restart to ensure
+  %{_bindir}/systemctl restart fapolicyd.service
+else
+  %{_sbindir}/fagenrules --load
+fi
```

Comment:
Looks like `%{_sbindir}/fagenrules --load` is still unconditional so we can simplify to:
```suggestion
%{_sbindir}/fagenrules --load
if %{_bindir}/systemctl is-active --quiet fapolicyd.service; then
  # loading fagenrules leaves fapolicyd.service in a dead state, restart to ensure
  %{_bindir}/systemctl restart fapolicyd.service
fi
```

Then for `is-active` and `restart` there's also `try-restart`

```suggestion
%{_sbindir}/fagenrules --load
# loading fagenrules leaves fapolicyd.service in a dead state, restart to ensure
%{_bindir}/systemctl try-restart fapolicyd.service
```

But then we don't benefit from modern helpers that only restart once in a single transaction that we have on modern EL (is that EL10+? At least it's already in Fedora). I have a feeling we should use `%systemd_postun_with_restart fapolicyd.service` instead